crates/wash-lib/src/build/component.rs

```diff
@@ -44,29 +42,37 @@ pub fn build_component(
         build_custom_component(common_config, component_config, raw_command)?
     } else {
         // Build component based on language toolchain
-        let component_wasm_path = match language_config {
+        match language_config {
             LanguageConfig::Rust(rust_config) => {
-                build_rust_component(common_config, rust_config, component_config)?
+                let rust_wasm_path =
+                    build_rust_component(common_config, rust_config, component_config)?;
+                if component_config.wasm_target == WasmTarget::WasiPreview2 {
+                    adapt_component_to_wasi_preview2(&rust_wasm_path, component_config)?
+                } else {
+                    rust_wasm_path
+                }
```

Comment:
SUPER NIT: I personally like a `match` here, mostly because I want it to be really obvious that we kind of make the "same" decision for p1 and modules, and then make a *different* one for p2.